csodiaq/identifier/targetedReanalysisFunctions.py

```diff
@@ -0,0 +1,232 @@
+from csodiaq.utils import format_protein_list_to_string
+import pandas as pd
+import numpy as np
+
+
+def calculate_mz_of_heavy_version_of_peptide(peptide, lightMz, z):
+    lightAndHeavyLysKMassDiff = 8.014199
+    lightAndHeavyArgRMassDiff = 10.00827
+
+    numLysK = peptide.count("K")
+    numArgR = peptide.count("R")
+
+    return (
+        lightMz
+        + (numLysK * lightAndHeavyLysKMassDiff) / z
+        + (numArgR * lightAndHeavyArgRMassDiff) / z
+    )
+
+
+def filter_to_only_keep_peptides_with_possibly_heavy_K_or_R_terminal_residue(fdrDf):
+    return fdrDf[
+        fdrDf["peptide"].str.endswith("R") | fdrDf["peptide"].str.endswith("K")
+    ].reset_index(drop=True)
+
+
+def filter_to_only_keep_top_peptides_unique_to_protein(fdrDf, maxPeptidesPerProtein):
+    fdrDf = (
+        fdrDf[fdrDf["uniquePeptide"] == 1]
+        .sort_values("ionCount", ascending=False)
+        .reset_index(drop=True)
+    )
+    return (
+        fdrDf.groupby(["leadingProtein"])
+        .head(maxPeptidesPerProtein)
+        .reset_index(drop=True)
+    )
+
+
+def calculate_mz_of_heavy_isotope_of_each_peptide(fdrDf):
+    return fdrDf.apply(
+        lambda x: calculate_mz_of_heavy_version_of_peptide(
+            x["peptide"], x["MzLIB"], x["zLIB"]
+        ),
+        axis=1,
+    )
+
+
+def make_bin_assignments_for_mz_values(mzValues, binWidth=0.75):
```

Comment:
Since the step inside `bins = [...]` is 2*binWidth, this seems like it's the bin half-width?